rskj-core/src/test/java/co/rsk/peg/BridgeSupportAddSignatureTest.java

```diff
@@ -534,6 +534,158 @@ void addSignatureMultipleInputsPartiallyValid() throws Exception {
+        TransactionOutput prevOut = new TransactionOutput(btcRegTestParams, prevTx, Coin.FIFTY_COINS, federation.getAddress());
+        prevTx.addOutput(prevOut);
+
+        BtcTransaction btcTx = new BtcTransaction(btcRegTestParams);
+        TransactionOutput output = new TransactionOutput(btcRegTestParams, btcTx, Coin.COIN, new BtcECKey().toAddress(btcRegTestParams));
+        btcTx.addOutput(output);
+        btcTx.addInput(prevOut).setScriptSig(createBaseInputScriptThatSpendsFromTheFederation(federation));
+
+        Sha256Hash unsignedBtxTxHash = btcTx.getHash();
+
+        ActivationConfig.ForBlock activations = mock(ActivationConfig.ForBlock.class);
+        when(activations.isActive(ConsensusRule.RSKIP298)).thenReturn(true);
+
+        SortedMap<Keccak256, BtcTransaction> rskTxWaitingForSignatures = new TreeMap<>();
+        rskTxWaitingForSignatures.put(keccak256, btcTx);
+        when(provider.getRskTxsWaitingForSignatures()).thenReturn(
+            rskTxWaitingForSignatures
+        );
+        when(provider.getPegoutCreationRskTxHashByBtcTxHash(unsignedBtxTxHash)).thenReturn(
+            Optional.of(keccak256)
+        );
+
+        BridgeSupport bridgeSupport = bridgeSupportBuilder
+            .withBridgeConstants(bridgeConstantsRegtest)
+            .withProvider(provider)
+            .withActivations(activations)
+            .build();
+
+        Script inputScript = btcTx.getInputs().get(0).getScriptSig();
+        List<ScriptChunk> chunks = inputScript.getChunks();
+        byte[] program = chunks.get(chunks.size() - 1).data;
+        Script redeemScript = new Script(program);
+        Sha256Hash sighash = btcTx.hashForSignature(0, redeemScript, BtcTransaction.SigHash.ALL, false);
+
+        BtcECKey.ECDSASignature sig = privateKeysToSignWith.get(0).sign(sighash);
+
+        byte[] derEncodedSig = sig.encodeToDER();
+
+        List derEncodedSigs = new ArrayList();
+        for (int i = 0; i < 1; i++) {
+            derEncodedSigs.add(derEncodedSig);
+        }
+
+        // Act
+        bridgeSupport.addSignature(findPublicKeySignedBy(federation.getBtcPublicKeys(), privateKeysToSignWith.get(0)), derEncodedSigs, keccak256.getBytes());
+
+        if (privateKeysToSignWith.size() > 1) {
+            BtcECKey.ECDSASignature sig2 = privateKeysToSignWith.get(1).sign(sighash);
+            byte[] derEncodedSig2 = sig2.encodeToDER();
+            List derEncodedSigs2 = new ArrayList();
+            for (int i = 0; i < 1; i++) {
+                derEncodedSigs2.add(derEncodedSig2);
+            }
+            bridgeSupport.addSignature(findPublicKeySignedBy(federation.getBtcPublicKeys(), privateKeysToSignWith.get(1)), derEncodedSigs2, keccak256.getBytes());
+        }
+
+        // Assert
+        verify(provider, atLeastOnce()).getPegoutCreationRskTxHashByBtcTxHash(unsignedBtxTxHash);
+        verify(provider, atLeastOnce()).setPegoutCreationEntry(
+            any()
+        );
```

Comment:
I can verify no calls are made to `getPegoutCreationRskTxHashByBtcTxHash` and `setPegoutCreationEntry` when still pending signatures to be added. But I cannot test the entry that exists in the index after adding one signature. Why? Because I'm testing `bridgeSupport`, not `bridgeStorageProvider`, for the `bridgeStorageProvider` I'm using a mock, and for that reason, that kind of test would be meaningless.